gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -298,20 +148,33 @@ public static SubscriberApi create(ServiceApiSettings<MethodIdentifier> settings
    * <!-- manual edit -->
    * <!-- end manual edit -->
    */
-  protected SubscriberApi(ServiceApiSettings<MethodIdentifier> settings) throws IOException {
-    this.settings = settings;
+  protected SubscriberApi(SubscriberSettings settings) throws IOException {
     this.channel = settings.getChannel();
 
-    Map<MethodIdentifier, ImmutableSet<Status.Code>> retryCodesConfig =
-        new EnumMap<>(DEFAULT_RETRY_CONFIG);
-    retryCodesConfig.putAll(settings.getRetryableCodes());
-    this.retryCodesConfig =
-        Maps.<MethodIdentifier, ImmutableSet<Status.Code>>immutableEnumMap(retryCodesConfig);
-
-    Map<MethodIdentifier, RetryParams> retryParamsConfig = new EnumMap<>(DEFAULT_RETRY_PARAMS);
-    retryParamsConfig.putAll(settings.getRetryParams());
-    this.retryParamsConfig =
-        Maps.<MethodIdentifier, RetryParams>immutableEnumMap(retryParamsConfig);
+    for (ApiCallSettings method : settings.allMethods()) {
```

Comment:
Hmm I think I made a mistake here. I was trying to make sure that each method doesn't create its own executor, but I think this code isn't successfully accomplishing that. There is a nuance about executors and channels since a channel needs an executor, but executors are needed for other things too. I fill fix it.